grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcService.java

```diff
@@ -221,19 +228,22 @@ private void frameAndServe(
             return;
         }
 
-        grpcResponse.aggregate().handleAsync(
+        grpcResponse.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc()).handle(
                 (framedResponse, t) -> {
-                    if (t != null) {
-                        res.completeExceptionally(t);
-                    } else {
-                        deframeAndRespond(ctx, framedResponse, res);
+                    try (SafeCloseable ignore = ctx.push()) {
+                        if (t != null) {
+                            PooledObjects.close(framedResponse.content());
```

Comment:
`framedResponse` might be null when t is not null.

Reply:
Nice catch!